### DISCUSSION TITLE
Notelock  shaking effect when hitting overlapping circle

### DISCUSSION THREAD

**Discussion by cerfeque** [ppy/osu#31091] [Q&A]
There is no way i am the first person who experienced this, so it's probably a known bug, but i think i should report anyway

https://github.com/user-attachments/assets/4ddb1971-136d-4d14-9844-b7c6274aa831

**Comment by dani211e** [comment]
Notelock still exists in lazer, it is just considerably more lenient. 
You tapped twice on the 2 stacked notes, causing one to be hit and the other to shake on the 2nd input. 
No bugs.

**Reply by cerfeque** [reply]
This is not the first time i see this behavior when i click the circle right in time when the other circle appears in the same place so i thought its a bug. But now when i looked at the replay at 0.30 speed i can see this is the double click. The only thing that bothers me is how short the input time was, i can't click this fast. Probably skill issue.

https://github.com/user-attachments/assets/b4d32caa-92d1-4e5c-89f0-9fecd7b84b77

**Comment by cerfeque** [comment]
Update: 
I still think this may be not my key press but the replay bug

Got the same encounter but this time im not pressing anything at all because im playing RELAX
I think this should be investigated

I think it happens only when you have the perfect timing of appear of the next circle on the place of the previous one, and it can not be recreated with autoplay, but it can be recreated with relax

https://github.com/user-attachments/assets/9699406b-ebde-4931-97f8-3fb283135682

**Reply by cihe13375** [reply]
Relax is just the program doing the clicks for you. When the note 3 shakes you can see a key press event on the key counter on the right side (k2 94->95), and since it's way too early the note just shakes instead of triggering the judgment. 

Indeed the program trying to click note 3 so early is a bug (https://github.com/ppy/osu/issues/27307), but in this case notelock saves it so you don't get two misses on 2 and 3.

**Reply by cerfeque** [reply]
And i think that actually there was no click, it shows that there was a click because game engine probably forced to show it. Something like when you click the note 1 and 3 simultaneously add 2 presses to the key counter because the game see 2 circles triggered and thinks that there were 2 presses and it need to show it, but there was only one click. Why am i thinking this way? You can look at previous clip where the same happened, but instead of relax i was playing with my hands, and if you look at how short time the key was pressed before release it makes me think this way, because no way human can do this, replay is already at 0.2x speed.

So, i think, if you get the right timing you can trigger 2 notes with one key press

Again, this happened to me always only in one scenario, when the upcoming circle spawns in the same place and right when you press the previous circle
And after i saw relax did the same thing i bet my ass that one click is triggering both circles and the press on the key counter is made up by the game


https://osu.ppy.sh/beatmapsets/1703935#osu
At the very end of this map you can find the pattern where you can most likely recreate this, i was able to do it multiple times
Im using dx11(experimental), maybe this matters, if someone decide to try recreating it too

**Comment by cerfeque** [comment]
My keypad sometimes double-clicking.
